src/Parser/Core/Modules/SpellHistory.js

```diff
@@ -42,6 +47,22 @@ class SpellHistory extends Analyzer {
     }
   }
 
+  _addTimeWaitingOnGCD(event) {
+    if (!this.lastGlobalCooldown) {
+      return event;
+    }
+    const resetBufferMS = RESET_BUFFER_PERCENT * this.lastGlobalCooldown.duration;
+    const earlyByMS = event.start + event.expectedDuration - event.timestamp;
+    // check if the ability was reset early. If the reset was less than a percentage of the GCD don't consider it as a reset.
```

Comment:
How would this handle Divine Purpose?

Reply:
It would work the same way as mangle (explained above)
before:
![image](https://user-images.githubusercontent.com/6773230/41685219-0cb05a5e-74e0-11e8-97c6-dde60b32f13c.png)
after:
![image](https://user-images.githubusercontent.com/6773230/41685239-1b42767e-74e0-11e8-844f-c13c85de9bc1.png)